### PR TITLE
Add CubeFile class (C++), a Py3JS renderer, and two tutorials on MO plotting and localization

### PR DESCRIPTION
**Pull request by fevangelista**
## Description
This PR adds a few new features for generating MO/density plots in Jupyter. This is still transient, so I am holding off writing manual pages.

## User Notes
- [x ] CubeFile class (C++) for fast(er) i/o
- [x] A Py3JS simple renderer (needs improvements)
- [x] Tutorials 05 and 06 on MO plotting and localization
- [x] Added new psi4/forte helpers (Python)
- [x] Reorganized the tutorial directory

## Checklist
- [x] Added tutorials for new features
- [x] Removed comments in code and input files
- [x] Documented source code
- [x] All short tests pass
- [x] Ready to go!

### CODE REVIEW

File: src/api/cube_file_api.cc [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+        .def(py::init<const std::string&>())
+        .def(py::init<const CubeFile&>())
+        .def("levels", &CubeFile::levels, "Get the isocontour levels to plot")
+        .def("num", &CubeFile::num, "Get the number of grid point in each direction")
+        .def("min", &CubeFile::min, "Get the minimum value of a coordinate")
+        .def("max", &CubeFile::max, "Get the maximum value of a coordinate")
+        .def("inc", &CubeFile::inc, "Get the increment in each direction")
+        .def("natoms", &CubeFile::natoms, "The number of atoms")
+        .def("atom_numbers", &CubeFile::atom_numbers, "The atomic numbers")
+        .def("atom_coords", &CubeFile::atom_coords, "The atomic numbers")
```
Comment (lcyyork) [inline src/api/cube_file_api.cc]: Forget to change the description?

File: src/api/cube_file_api.cc [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+        .def(
+            "data", [](CubeFile& cf) { return vector_to_np(cf.data(), cf.num()); }, "Get the data")
+        .def("compute_levels", &CubeFile::compute_levels,
+             "Compute the isolevel that encompasses a give fraction of the total density")
+        .def("scale", &CubeFile::scale, "The atomic numbers")
+        .def("add", &CubeFile::add, "The atomic numbers")
```
Comment (lcyyork) [inline src/api/cube_file_api.cc]: Same for these three lines. I guess they can be left blank.

File: src/base_classes/forte_options.cc [hunk 3]
```diff
@@ -221,6 +233,90 @@ std::vector<double> ForteOptions::get_double_vec(const std::string& label) {
...
+void set_gen_list(const std::string& label, py::list val);
+void set_int_vec(const std::string& label);
+void set_double_vec(const std::string& label, const std::vector<double>& val);
+
+void ForteOptions::set_bool(const std::string& label, bool val) {
+    auto value_type = get(label);
+    if (value_type.second == "bool") {
+        set(label, py::bool_(val));
+    } else {
+        std::string msg = "Called ForteOptions::set_bool(" + label +
```
Comment (lcyyork) [inline src/base_classes/forte_options.cc]: I wonder if these error messages along with those in the get_… functions can be standardized into one function.

File: src/base_classes/forte_options.h [hunk 4]
```diff
@@ -187,6 +194,55 @@ class ForteOptions {
...
 
+    /**
+     * @brief Set a boolean option
+     * @param label Option label
+     * @param val Option value
+     */
+    void set_bool(const std::string& label, bool val);
+
+    /**
+     * @brief Get a integer option
```
Comment (lcyyork) [inline src/base_classes/forte_options.h]: The briefs should be Set here and after.

File: src/helpers/helpers.h [hunk 5]
```diff
@@ -61,6 +61,9 @@ namespace forte {
  */
 py::array_t<double> ambit_to_np(ambit::Tensor t);
 
+py::array_t<double> vector_to_np(const std::vector<double>& v, const std::vector<size_t>& dims);
```
Comment (lcyyork) [inline src/helpers/helpers.h]: Add comments.

File: src/api/cube_file_api.cc (hunk 1 above)
Comment (fevangelista) [reply src/api/cube_file_api.cc]: I put them in now.

File: src/api/cube_file_api.cc (hunk 2 above)
Comment (fevangelista) [reply src/api/cube_file_api.cc]: I put them in now.

File: src/base_classes/forte_options.cc (hunk 3 above)
Comment (fevangelista) [reply src/base_classes/forte_options.cc]: Good point, done.

File: src/base_classes/forte_options.h (hunk 4 above)
Comment (fevangelista) [reply src/base_classes/forte_options.h]: Thanks for catching this!

**Comment by fevangelista** [comment]
@lcyyork: this is ready for merge. Test out the tutorials 05, 05b, and 06. The python code is super documented now.

**Pull request merged**